server/src/termdb.server.init.js

```diff
@@ -473,93 +473,102 @@ export function server_init_db_queries(ds) {
 	returns list of chart types based on term types from each subcohort combination
 	*/
 	q.getSupportedChartTypes = embedder => {
-		mayComputeTermtypeByCohort(ds)
-		// ds.cohort.termdb.termtypeByCohort[] is set
+		const forbiddenRoutes = authApi.getForbiddenRoutesForDsEmbedder(ds.label, embedder)
 
-		const cred = serverconfig.dsCredentials?.[ds.label]
-		const supportedChartTypes = {}
-		const numericTypeCount = {}
-		// key: subcohort combinations, comma-joined, alphabetically sorted, as in the subcohort_terms table
-		// value: array of chart types allowed by term types
-		for (const r of ds.cohort.termdb.termtypeByCohort) {
-			if (!r.type) continue // skip ungraphable parent terms
-
-			if (!(r.cohort in supportedChartTypes)) {
-				supportedChartTypes[r.cohort] = new Set(['regression', 'summary', 'facet'])
-				if (ds.cohort.scatterplots) supportedChartTypes[r.cohort].add('sampleScatter')
-
-				numericTypeCount[r.cohort] = 0
-				if (ds.cohort.allowedChartTypes?.includes('matrix')) supportedChartTypes[r.cohort].add('matrix')
-				if (ds.cohort.allowedChartTypes?.includes('brainImaging')) supportedChartTypes[r.cohort].add('brainImaging')
-				if (ds.cohort.allowedChartTypes?.includes('numericDictTermCluster'))
-					supportedChartTypes[r.cohort].add('numericDictTermCluster')
-				const forbiddenRoutes = authApi.getForbiddenRoutesForDsEmbedder(ds.label, embedder)
-				if (!forbiddenRoutes.includes('termdb') && !forbiddenRoutes.includes('*')) {
-					supportedChartTypes[r.cohort].add('dataDownload')
-					supportedChartTypes[r.cohort].add('sampleView')
-					if (ds.queries?.singleCell) supportedChartTypes[r.cohort].add('singleCellPlot')
-				}
-			}
-			if (serverconfig.features?.draftChartTypes) {
-				// TODO: move draft charts out of flag once stable
-				supportedChartTypes[r.cohort].add(...serverconfig.features.draftChartTypes)
-			}
-			if (r.type == 'survival' && !supportedChartTypes[r.cohort].has('survival'))
-				supportedChartTypes[r.cohort].add('survival')
-			if (r.type == 'condition' && !supportedChartTypes[r.cohort].has('cuminc'))
-				supportedChartTypes[r.cohort].add('cuminc')
-			if (numericTypes.has(r.type)) numericTypeCount[r.cohort] += r.samplecount
-		}
+		const isSupportedChart = {
```

Comment:
this is a good design. it can be improved by removing profile charts. this should only define general chart types that can be determined by term type availability. profile charts and any other special purpose charts are only defined in their own ds.cohort.specialCharts[]. and will be added to `supportedChartTypes` as-is.

Reply:
So that would mean that any chartType that's listed in `specialCharts`, is not meant to be be listed in `hiddenCharts`, right?

Reply:
Maybe we can have a method that does a switch with cases for the charts with more complex logic and a default behaviour, something like:
`
isSupported(chart)
{
switch(chart):
...

default:
return specialCharts.includes(chart) && !hiddenCharts.includes(chart)
}
`